lib/oidcstrategy.js

```diff
@@ -207,350 +299,81 @@ Strategy.prototype.authenticate = function authenticateStrategy(req, options) {
           // @TODO: could be undefined
           const policy = req.query.p;
 
-          // @TODO: assumes that self._options.identityMetadata is of type string.
-          //        according to Strategy constructor, it could also be a PEM encodede public key
-          metadata = self._options.identityMetadata
+          metadataUrl = self._options.identityMetadata
             .replace('common', self._options.tenantName)
             .concat(`?p=${policy}`);
 
-          self._cacheKey = policy; // this policy will become cache key.
-
-          log.info('B2C: New Metadata url provided to Strategy was: ', metadata);
-        }
+          cachekey = 'policy: ' + policy; // this policy will become cache key.
```

Comment:
On our team we've been moving to using the newer template strings in ES6, so this would be:

```
cacheKey = `policy: ${policy}`'
```

Just something to consider but just a nit.